diff --git a/tests/sentry/relay/snapshots/test_config/test_get_project_config/REGION.pysnap b/tests/sentry/relay/snapshots/test_config/test_get_project_config/REGION.pysnap
index 4c81c60c..3e7bd783 100644
--- a/tests/sentry/relay/snapshots/test_config/test_get_project_config/REGION.pysnap
+++ b/tests/sentry/relay/snapshots/test_config/test_get_project_config/REGION.pysnap
@@ -179,9 +179,14 @@ config:
   performanceScore:
     profiles:
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Chrome
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Chrome
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Chrome
+        op: or
       name: Chrome
       scoreComponents:
       - measurement: fcp
@@ -205,9 +210,14 @@ config:
         p50: 400.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Firefox
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Firefox
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Firefox
+        op: or
       name: Firefox
       scoreComponents:
       - measurement: fcp
@@ -231,9 +241,14 @@ config:
         p50: 400.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Safari
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Safari
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Safari
+        op: or
       name: Safari
       scoreComponents:
       - measurement: fcp
@@ -257,9 +272,14 @@ config:
         p50: 400.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Edge
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Edge
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Edge
+        op: or
       name: Edge
       scoreComponents:
       - measurement: fcp
@@ -283,9 +303,14 @@ config:
         p50: 400.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Opera
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Opera
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Opera
+        op: or
       name: Opera
       scoreComponents:
       - measurement: fcp
@@ -316,6 +341,12 @@ config:
         - name: event.contexts.browser.name
           op: eq
           value: Google Chrome
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Chrome
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Google Chrome
         op: or
       name: Chrome INP
       scoreComponents:
@@ -325,9 +356,14 @@ config:
         p50: 500.0
         weight: 1.0
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Edge
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Edge
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Edge
+        op: or
       name: Edge INP
       scoreComponents:
       - measurement: inp
@@ -336,9 +372,14 @@ config:
         p50: 500.0
         weight: 1.0
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Opera
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Opera
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Opera
+        op: or
       name: Opera INP
       scoreComponents:
       - measurement: inp
@@ -347,9 +388,14 @@ config:
         p50: 500.0
         weight: 1.0
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Chrome Mobile
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Chrome Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Chrome Mobile
+        op: or
       name: Chrome Mobile
       scoreComponents:
       - measurement: fcp
@@ -373,9 +419,14 @@ config:
         p50: 1800.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Firefox Mobile
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Firefox Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Firefox Mobile
+        op: or
       name: Firefox Mobile
       scoreComponents:
       - measurement: fcp
@@ -399,9 +450,14 @@ config:
         p50: 1800.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Mobile Safari
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Mobile Safari
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Mobile Safari
+        op: or
       name: Safari Mobile
       scoreComponents:
       - measurement: fcp
@@ -425,9 +481,14 @@ config:
         p50: 1800.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Edge Mobile
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Edge Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Edge Mobile
+        op: or
       name: Edge Mobile
       scoreComponents:
       - measurement: fcp
@@ -451,9 +512,14 @@ config:
         p50: 1800.0
         weight: 0.1
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Opera Mobile
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Opera Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Opera Mobile
+        op: or
       name: Opera Mobile
       scoreComponents:
       - measurement: fcp
@@ -481,6 +547,9 @@ config:
         - name: event.contexts.browser.name
           op: eq
           value: Chrome Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Chrome Mobile
         op: or
       name: Chrome Mobile INP
       scoreComponents:
@@ -490,9 +559,14 @@ config:
         p50: 500.0
         weight: 1.0
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Edge Mobile
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Edge Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Edge Mobile
+        op: or
       name: Edge Mobile INP
       scoreComponents:
       - measurement: inp
@@ -501,9 +575,14 @@ config:
         p50: 500.0
         weight: 1.0
     - condition:
-        name: event.contexts.browser.name
-        op: eq
-        value: Opera Mobile
+        inner:
+        - name: event.contexts.browser.name
+          op: eq
+          value: Opera Mobile
+        - name: span.attributes.browser.name.value
+          op: eq
+          value: Opera Mobile
+        op: or
       name: Opera Mobile INP
       scoreComponents:
       - measurement: inp
diff --git a/tests/sentry/relay/test_config.py b/tests/sentry/relay/test_config.py
index 407a720d..ac1ded45 100644
--- a/tests/sentry/relay/test_config.py
+++ b/tests/sentry/relay/test_config.py
@@ -26,7 +26,6 @@ from sentry.snuba.dataset import Dataset
 from sentry.testutils.factories import Factories
 from sentry.testutils.helpers import Feature
 from sentry.testutils.helpers.datetime import freeze_time
-from sentry.testutils.helpers.options import override_options
 from sentry.testutils.pytest.fixtures import InstaSnapshotter, django_db_all
 from sentry.testutils.silo import cell_silo_test
 from sentry.utils.safe import get_path
@@ -97,6 +96,15 @@ def _validate_project_config(config):
     assert normalize_project_config(config) == config
 
 
+def _canonicalize_performance_profile_conditions(config) -> None:
+    for profile in config.get("performanceScore", {}).get("profiles", []):
+        condition = profile.get("condition", {})
+        if condition.get("op") == "or":
+            condition["inner"].sort(
+                key=lambda item: (item.get("name"), item.get("value"), item.get("op"))
+            )
+
+
 @django_db_all
 @cell_silo_test
 def test_get_project_config_non_visible(default_project) -> None:
@@ -118,6 +126,7 @@ def test_get_project_config(default_project: Project, insta_snapshot: InstaSnaps
     cfg = project_cfg.to_dict()
 
     _validate_project_config(cfg["config"])
+    _canonicalize_performance_profile_conditions(cfg["config"])
 
     # Remove keys that change everytime
     cfg.pop("lastChange")
@@ -752,6 +761,7 @@ def test_desktop_performance_calculate_score(default_project) -> None:
         profile["version"] = "1"
 
     assert normalize_project_config(config) == config
+    _canonicalize_performance_profile_conditions(config)
     performance_score = config["performanceScore"]["profiles"]
     assert performance_score[0] == {
         "name": "Chrome",
@@ -786,9 +796,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Chrome",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Chrome",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Chrome",
+                },
+            ],
         },
         "version": "1",
     }
@@ -825,9 +845,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Firefox",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Firefox",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Firefox",
+                },
+            ],
         },
         "version": "1",
     }
@@ -864,9 +894,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Safari",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Safari",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Safari",
+                },
+            ],
         },
         "version": "1",
     }
@@ -903,9 +943,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Edge",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Edge",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Edge",
+                },
+            ],
         },
         "version": "1",
     }
@@ -942,9 +992,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Opera",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Opera",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Opera",
+                },
+            ],
         },
         "version": "1",
     }
@@ -973,6 +1033,16 @@ def test_desktop_performance_calculate_score(default_project) -> None:
                     "name": "event.contexts.browser.name",
                     "value": "Google Chrome",
                 },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Chrome",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Google Chrome",
+                },
             ],
         },
         "version": "1",
@@ -990,9 +1060,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Edge",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Edge",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Edge",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1009,9 +1089,19 @@ def test_desktop_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Opera",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Opera",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Opera",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1026,6 +1116,7 @@ def test_mobile_performance_calculate_score(default_project) -> None:
         profile["version"] = "1"
 
     assert normalize_project_config(config) == config
+    _canonicalize_performance_profile_conditions(config)
     performance_score = config["performanceScore"]["profiles"]
 
     assert performance_score[8] == {
@@ -1061,9 +1152,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Chrome Mobile",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Chrome Mobile",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Chrome Mobile",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1100,9 +1201,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Firefox Mobile",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Firefox Mobile",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Firefox Mobile",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1139,9 +1250,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Mobile Safari",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Mobile Safari",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Mobile Safari",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1178,9 +1299,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Edge Mobile",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Edge Mobile",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Edge Mobile",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1218,9 +1349,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Opera Mobile",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Opera Mobile",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Opera Mobile",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1243,6 +1384,11 @@ def test_mobile_performance_calculate_score(default_project) -> None:
                     "name": "event.contexts.browser.name",
                     "value": "Chrome Mobile",
                 },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Chrome Mobile",
+                },
             ],
         },
         "version": "1",
@@ -1259,9 +1405,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             },
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Edge Mobile",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Edge Mobile",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Edge Mobile",
+                },
+            ],
         },
         "version": "1",
     }
@@ -1277,9 +1433,19 @@ def test_mobile_performance_calculate_score(default_project) -> None:
             }
         ],
         "condition": {
-            "op": "eq",
-            "name": "event.contexts.browser.name",
-            "value": "Opera Mobile",
+            "op": "or",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "event.contexts.browser.name",
+                    "value": "Opera Mobile",
+                },
+                {
+                    "op": "eq",
+                    "name": "span.attributes.browser.name.value",
+                    "value": "Opera Mobile",
+                },
+            ],
         },
         "version": "1",
     }
