diff --git a/tests/Feature/EnumsTest.php b/tests/Feature/EnumsTest.php
index 14b871b..5e62f62 100644
--- a/tests/Feature/EnumsTest.php
+++ b/tests/Feature/EnumsTest.php
@@ -80,8 +80,8 @@ it('captures unit enum cases correctly', function () {
 
     expect($userRole->cases)->toHaveCount(2);
     expect($userRole->cases)->toMatchArray([
-        'BAR' => null,
-        'FOO' => null,
+        'BAR' => 0,
+        'FOO' => 1,
     ]);
 });
 
