diff --git a/railties/test/generators/app_generator_test.rb b/railties/test/generators/app_generator_test.rb
index 767f417bc6..fc7550aef7 100644
--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
@@ -768,6 +768,13 @@ def test_inclusion_of_kamal_files
     assert_file ".kamal/secrets"
   end
 
+  def test_inclusion_of_bootsnap_files
+    generator [destination_root]
+    run_generator_instance
+
+    assert_file "config/bootsnap.rb"
+  end
+
   def test_kamal_files_are_skipped_if_required
     generator [destination_root], ["--skip-kamal"]
     run_generator_instance
