diff --git a/test/e2e-cypress/e2e/features/try-it-out-schema-type-array-example-type-string.cy.js b/test/e2e-cypress/e2e/features/try-it-out-schema-type-array-example-type-string.cy.js
new file mode 100644
index 00000000..cf25a9c0
--- /dev/null
+++ b/test/e2e-cypress/e2e/features/try-it-out-schema-type-array-example-type-string.cy.js
@@ -0,0 +1,12 @@
+describe("Try it out with schema type array but example type string", () => {
+    it("shows a validation error message when Execute is clicked", () => {
+      cy
+        .visit("?tryItOutEnabled=true&url=/documents/features/try-it-out-schema-type-array-example-type-string.yaml")
+        .get("#operations-default-get_")
+        .click()
+        .get(".btn.execute")
+        .click()
+        .get(".validation-errors")
+        .should("exist")
+    })
+  })
diff --git a/test/e2e-cypress/static/documents/features/try-it-out-schema-type-array-example-type-string.yaml b/test/e2e-cypress/static/documents/features/try-it-out-schema-type-array-example-type-string.yaml
new file mode 100644
index 00000000..118afd1e
--- /dev/null
+++ b/test/e2e-cypress/static/documents/features/try-it-out-schema-type-array-example-type-string.yaml
@@ -0,0 +1,23 @@
+openapi: 3.0.3
+info:
+  title: test
+  version: 1.0.0
+paths:
+  /:
+    get:
+      parameters:
+        - in: query
+          name: test
+          required: true
+          schema:
+            type: array
+            example: 'test1'
+            items:
+              type: string
+              enum:
+                - 'test1'
+                - 'test2'
+                - 'test3'
+      responses:
+        default:
+          description: ok
diff --git a/test/unit/core/utils.js b/test/unit/core/utils.js
index 5787fd7d..f6b7d9d0 100644
--- a/test/unit/core/utils.js
+++ b/test/unit/core/utils.js
@@ -350,11 +350,21 @@ describe("utils", () => {
     let value = null
     let result = null
 
-    const assertValidateParam = (param, value, expectedError) => {
+    const assertValidateOas3Param = (param, value, expectedError) => {
+      // for cases where you _only_ want to try OAS3
+      result = validateParam(fromJS(param), value, {
+        isOAS3: true
+      })
+      expect( result ).toEqual( expectedError )
+    }
+
+    const assertValidateOas2Param = (param, value, expectedError) => {
       // Swagger 2.0 version
       result = validateParam( fromJS(param), fromJS(value))
       expect( result ).toEqual( expectedError )
+    }
 
+    const assertValidateOas3ParamWithSchema = (param, value, expectedError) => {
       // OAS3 version, using `schema` sub-object
       let oas3Param = {
         required: param.required,
@@ -363,18 +373,14 @@ describe("utils", () => {
           required: undefined
         }
       }
-      result = validateParam( fromJS(oas3Param), fromJS(value), {
-        isOAS3: true
-      })
-      expect( result ).toEqual( expectedError )
+      assertValidateOas3Param(oas3Param, value, expectedError)
     }
 
-    const assertValidateOas3Param = (param, value, expectedError) => {
-      // for cases where you _only_ want to try OAS3
-      result = validateParam(fromJS(param), value, {
-        isOAS3: true
-      })
-      expect( result ).toEqual( expectedError )
+    const assertValidateParam = (param, value, expectedError) => {
+      // Swagger 2.0 version
+      assertValidateOas2Param(param, value, expectedError)
+      // OAS3 version, using `schema` sub-object
+      assertValidateOas3ParamWithSchema(param, value, expectedError)
     }
 
     it("should check the isOAS3 flag when validating parameters", () => {
@@ -749,7 +755,8 @@ describe("utils", () => {
         type: "array"
       }
       value = "[1]"
-      assertValidateParam(param, value, [])
+      assertValidateOas3ParamWithSchema(param, value, expect.arrayContaining([expect.any(String)]))
+      assertValidateOas2Param(param, value, [])
 
       // valid array, items match type
       param = {
