diff --git a/packages/svelte/tests/runtime-runes/samples/async-error-in-script/_config.js b/packages/svelte/tests/runtime-runes/samples/async-error-in-script/_config.js
new file mode 100644
index 000000000..2e0fab473
--- /dev/null
+++ b/packages/svelte/tests/runtime-runes/samples/async-error-in-script/_config.js
@@ -0,0 +1,6 @@
+import { test } from '../../test';
+
+export default test({
+	mode: ['async-server', 'hydrate'],
+	error: 'oops'
+});
diff --git a/packages/svelte/tests/runtime-runes/samples/async-error-in-script/main.svelte b/packages/svelte/tests/runtime-runes/samples/async-error-in-script/main.svelte
new file mode 100644
index 000000000..64d9e6ce0
--- /dev/null
+++ b/packages/svelte/tests/runtime-runes/samples/async-error-in-script/main.svelte
@@ -0,0 +1,7 @@
+<script>
+	await 1;
+
+	throw new Error('oops');
+</script>
+
+<h1>hello</h1>
