diff --git a/src/Workspaces/CoreTest/WorkspaceTests/AdhocWorkspaceTests.cs b/src/Workspaces/CoreTest/WorkspaceTests/AdhocWorkspaceTests.cs
index d6a71b46fb4..f712bd9d745 100644
--- a/src/Workspaces/CoreTest/WorkspaceTests/AdhocWorkspaceTests.cs
+++ b/src/Workspaces/CoreTest/WorkspaceTests/AdhocWorkspaceTests.cs
@@ -592,4 +592,18 @@ public void TestNotGCRootedOnConstruction()
         // rooted within the export provider instance.
         GC.KeepAlive(exportProvider);
     }
+
+    [Fact]
+    public async Task VbReferencingCSharp()
+    {
+        var workspace = new AdhocWorkspace();
+        var csProj = workspace.AddProject("CsProj", LanguageNames.CSharp);
+        csProj = csProj.WithCompilationOptions(csProj.CompilationOptions.WithOutputKind(OutputKind.DynamicallyLinkedLibrary));
+        var vbProj = csProj.Solution.AddProject("VbProj", "VbProj", LanguageNames.VisualBasic)
+            .AddProjectReference(new ProjectReference(csProj.Id));
+        vbProj = vbProj.WithCompilationOptions(vbProj.CompilationOptions.WithOutputKind(OutputKind.DynamicallyLinkedLibrary));
+        var comp = (await vbProj.GetCompilationAsync())!;
+        var diagnostics = comp.GetDiagnostics();
+        Assert.Empty(diagnostics);
+    }
 }
