diff --git a/test/fixtures/assets/first_path/test.html b/test/fixtures/assets/first_path/test.html
new file mode 100644
index 0000000..76c3ac3
--- /dev/null
+++ b/test/fixtures/assets/first_path/test.html
@@ -0,0 +1 @@
+<html><body>hello</body></html>
diff --git a/test/propshaft/asset_test.rb b/test/propshaft/asset_test.rb
index a5f024a..037c199 100644
--- a/test/propshaft/asset_test.rb
+++ b/test/propshaft/asset_test.rb
@@ -19,6 +19,13 @@ class Propshaft::AssetTest < ActiveSupport::TestCase
     assert_equal "text/css", find_asset("another.css").content_type.to_s
   end
 
+  test "content type with charset" do
+    assert_equal "text/plain", find_asset("one.txt").content_type_with_charset.to_s
+    assert_equal "text/javascript", find_asset("again.js").content_type_with_charset.to_s
+    assert_equal "text/css; charset=utf-8", find_asset("another.css").content_type_with_charset.to_s
+    assert_equal "text/html; charset=utf-8", find_asset("test.html").content_type_with_charset.to_s
+  end
+
   test "length" do
     assert_equal 19, find_asset("one.txt").length
   end
diff --git a/test/propshaft/server_test.rb b/test/propshaft/server_test.rb
index 6db5581..51d175d 100644
--- a/test/propshaft/server_test.rb
+++ b/test/propshaft/server_test.rb
@@ -47,7 +47,7 @@ class Propshaft::ServerTest < ActiveSupport::TestCase
 
     assert_equal 200, last_response.status
     assert_equal last_response.body.bytesize.to_s, last_response.headers['content-length']
-    assert_equal "text/css", last_response.headers['content-type']
+    assert_equal "text/css; charset=utf-8", last_response.headers['content-type']
     assert_equal "Accept-Encoding", last_response.headers['vary']
     assert_equal "\"#{asset.digest}\"", last_response.headers['etag']
     assert_equal "public, max-age=31536000, immutable", last_response.headers['cache-control']
@@ -67,6 +67,22 @@ class Propshaft::ServerTest < ActiveSupport::TestCase
     assert_equal 200, last_response.status
   end
 
+  test "serve an HTML file with charset" do
+    asset = @assembly.load_path.find("test.html")
+    get "/assets/#{asset.digested_path}"
+
+    assert_equal 200, last_response.status
+    assert_equal "text/html; charset=utf-8", last_response.headers['content-type']
+  end
+
+  test "serve a JS file without charset" do
+    asset = @assembly.load_path.find("again.js")
+    get "/assets/#{asset.digested_path}"
+
+    assert_equal 200, last_response.status
+    assert_equal "text/javascript", last_response.headers['content-type']
+  end
+
   test "not found" do
     get "/assets/not-found.js"
 
