diff --git a/x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/search/utils/get_indices_for_product_names.test.ts b/x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/search/utils/get_indices_for_product_names.test.ts
index 8625ce9f521..3a02858bcd8 100644
--- a/x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/search/utils/get_indices_for_product_names.test.ts
+++ b/x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/search/utils/get_indices_for_product_names.test.ts
@@ -21,7 +21,7 @@ describe('getIndicesForProductNames', () => {
     ]);
   });
   it('returns individual index names when ELSER EIS is specified', () => {
-    expect(getIndicesForProductNames(['kibana', 'elasticsearch'], '.elser-v2-elastic')).toEqual([
+    expect(getIndicesForProductNames(['kibana', 'elasticsearch'], '.elser-2-elastic')).toEqual([
       getProductDocIndexName('kibana'),
       getProductDocIndexName('elasticsearch'),
     ]);
